src/test/java/com/revature/studyforce/flashcard/integration/VoteIntegrationTest.java

```diff
@@ -0,0 +1,162 @@
+        userRepository.save(u);
+
+        Flashcard flashcard = new Flashcard(0,u,null,"how is your day",1,1,null,null,false);
+        flashcardRepository.save(flashcard);
+
+        Answer a = new Answer(0,u,flashcard,"check stackoverflow",5,false,false,Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
+        answerRepository.save(a);
+
+        MockMvc mockMvc = MockMvcBuilders.standaloneSetup(controller).build();
+        String string = "{\"answerId\" : \"3\","
+                + "\"userId\" : \"1\","
+                + "\"value\" : \"1\""
+                + "}";
+        System.out.println(string);
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/flashcards/votes/")
+                .content(string)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        System.out.println(result.getResponse().getContentAsString());
+        System.out.println(result.getResponse().getStatus());
+    }
+
+    @Test
+    void givenVoteWithBadValue_whenCreateAnswerVote_shouldThrowException() throws Exception {
+        User u = new User(0,"jesus.christ@revature.com","password","Jesus","Christ",true,false,false, Authority.USER, Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
+        userRepository.save(u);
+
+        Flashcard flashcard = new Flashcard(0,u,null,"how is your day",1,1,null,null,false);
+        flashcardRepository.save(flashcard);
+
+        Answer a = new Answer(0,u,flashcard,"check stackoverflow",5,false,false,Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
+        answerRepository.save(a);
+
+        MockMvc mockMvc = MockMvcBuilders.standaloneSetup(controller).build();
+        String string = "{\"answerId\" : \"3\","
+                + "\"userId\" : \"1\","
+                + "\"value\" : \"33\""
+                + "}";
+        System.out.println(string);
+        MvcResult result = mockMvc.perform(MockMvcRequestBuilders.post("/flashcards/votes/")
+                .content(string)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().is4xxClientError())// .content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
+                .andReturn();
+
+        Assertions.assertEquals("Invalid vote value exception", result.getResponse().getErrorMessage());
+
+    }
+
+    @Test
+    void givenVoteWithBadUserId_whenCreateAnswerVote_shouldThrowException() throws Exception {
+        User u = new User(0,"jesus.christ@revature.com","password","Jesus","Christ",true,false,false, Authority.USER, Timestamp.valueOf(LocalDateTime.now()),Timestamp.valueOf(LocalDateTime.now()));
```

Comment:
Using outdated user model, please correct to user model with 1 name field and no password field